storage/scanner.go

```diff
@@ -170,8 +175,50 @@ func (rs *rangeScanner) paceInterval(start, now time.Time) time.Duration {
 	return interval
 }
 
+// waitAndProcess waits for the pace interval and processes the range
+// if rng is not nil. The methor returns true when the scanner needs
+// to be stopped. The method also removes a range from queues when it
+// is signaled via the removed channel.
+func (rs *rangeScanner) waitAndProcess(start time.Time, clock *hlc.Clock, stopper *util.Stopper,
+	stats *storeStats, rng *Range) bool {
+	waitInterval := rs.paceInterval(start, time.Now())
+	if log.V(6) {
+		log.Infof("Wait time interval set to %s", waitInterval)
+	}
+	for {
+		select {
+		case <-time.After(waitInterval):
```

Comment:
Construct the time.After channel outside the loop, so the delay doesn't reset every time a new range comes in on rs.removed.